pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -111,18 +113,93 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 // Name returns Name for the precondition.
 func (pf *Upgradeable) Name() string { return "ClusterVersionUpgradeable" }
 
+//  RecentEtcdBackup checks if a recent etcd backup has been taken.
+type RecentEtcdBackup struct {
+	key      string
+	lister   configv1listers.ClusterVersionLister
+	coLister configv1listers.ClusterOperatorLister
+}
+
+// NewRecentEtcdBackup returns a new RecentEtcdBackup precondition check.
+func NewRecentEtcdBackup(lister configv1listers.ClusterVersionLister, coLister configv1listers.ClusterOperatorLister) *RecentEtcdBackup {
+	return &RecentEtcdBackup{
+		key:      "version",
+		lister:   lister,
+		coLister: coLister,
+	}
+}
+
+func recentEtcdBackupCondition(lister configv1listers.ClusterOperatorLister) (string, string) {
+	var reason, message string
+	ops, err := lister.Get("etcd")
+	if err == nil {
+		backupCondition := resourcemerge.FindOperatorStatusCondition(ops.Status.Conditions, "RecentBackup")
+		if backupCondition == nil {
+			reason = "EtcdRecentBackupNotSet"
+			message = "RecentBackup: Etcd backup condition is not set."
```

Comment:
nit: "Etcd" -> "etcd".  They prefer downcased, even when they're the first word in a sentence, e.g. [here][1].

[1]: https://github.com/etcd-io/etcd#etcd

Reply:
done

Reply:
For context: The root is derived from the /etc directory.